File: Ruby/test/Rebill/Create_Recurring_Payment_CC.rb
##
# BluePay Ruby Sample code.
#
# This code sample creates a recurring payment charging $15.00 per month for one year.

require_relative "../../lib/bluepay.rb"

acct_id = "Merchant's Account ID Here"
sec_key = "Merchant's Secret Key Here"
trans_mode = "TEST" # Transaction Mode (can also be "LIVE")

rebill = BluePay.new(
  account_id: acct_id,   
  secret_key: sec_key,   
  mode: trans_mode
)

rebill.set_customer_information(
  first_name: "Bob", 
  last_name: "Tester",
  address1: "123 Test St.", 
  address2: "Apt #500", 
  city: "Testville", 
  state: "IL", 
  zip_code: "54321", 
  country: "USA",
  phone: "123-123-1234",  
  email: "test@bluepay.com"  
)

rebill.set_cc_information(
  cc_number: "4111111111111111", # Customer Credit Card Number
  cc_expiration: "1215", # Card Expiration Date: MMYY
  cvv2: "123" # Card CVV2
)

rebill.set_recurring_payment(
  reb_first_date: "2015-01-01", # Rebill Start Date: Jan. 1, 2015
  reb_expr: "1 MONTH", # Rebill Frequency: 1 MONTH
  reb_cycles: "12", # Rebill # of Cycles: 12
  reb_amount: "15.00" # Rebill Amount: $15.00
)

rebill.auth(amount: "0.00") 

# Makes the API Request
rebill.process

# if transaction was successful
if rebill.successful_response?
  # Reads the response from BluePay
  puts "TRANSACTION ID: " + rebill.get_trans_id
  puts "REBILL ID: " + rebill.get_rebill_id
  puts "TRANSACTION STATUS: " + rebill.get_status
  puts "AVS RESPONSE: " + rebill.get_avs_code
  puts "CVV2 RESPONSE: " + rebill.get_cvv2_code
  puts "MASKED PAYMENT ACCOUNT: " + rebill.get_masked_account
  puts "CARD TYPE: " + rebill.get_card_type
  puts "AUTH CODE: " + rebill.get_auth_code
else
  puts rebill.get_message
end

File: VB/vbnet/Rebill/Create_Recurring_Payment_CC.vb
' *
' * Bluepay VB.NET Sample code.
' *
' * This code sample creates a recurring rebill charging $15.00 per month for one year.
' * See comments below on the details of the initial Setup of the
' * rebilling cycle.
' * If using TEST mode, odd dollar amounts will return
' * an approval and even dollar amounts will return a decline.
' *

Imports System
Imports vbnet.BPVB

Namespace Rebill

    Public Class CreateRecurringPaymentCC

        Public Shared Sub run()

            Dim accountID As String = "Merchant's Account ID Here"
            Dim secretKey As String = "Merchant's Secret Key Here"
            Dim mode As String = "TEST" 

            Dim rebill As BluePay = New BluePay(
                accountID,
                secretKey,
                mode
            )

            rebill.setCustomerInformation(
                firstName:="Bob",
                lastName:="Tester",
                address1:="1234 Test St.",
                address2:="Apt #500",
                city:="Testville",
                state:="IL",
                zipCode:="54321", 
                country:="USA",
                phone:="123-123-12345",
                email:="test@bluepay.com"
            )

            rebill.setCCInformation(
                ccNumber:="4111111111111111", 
                ccExpiration:="1215", 
                cvv2:="123" 
            )

            ' Rebill Start Date: YYYY-MM-DD
            ' Rebill Frequency: 1 MONTH
            ' Rebill number of Cycles: 12
            ' Rebill Amount: $15.00
            rebill.setRebillingInformation(
                rebFirstDate:="2015-01-01", 
                rebExpr:="1 MONTH", 
                rebCycles:="12", 
                rebAmount:="15.00" 
            )

            rebill.auth(amount:="0.00")

            rebill.process()

            If rebill.isSuccessfulTransaction() Then
                Console.Write("Transaction ID: " + rebill.getTransID() + Environment.NewLine)
                Console.Write("Rebill ID: " + rebill.getRebillID() + Environment.NewLine)
                Console.Write("Transaction Status: " + rebill.getStatus() + Environment.NewLine)
                Console.Write("Transaction Message: " + rebill.getMessage() + Environment.NewLine)
                Console.Write("AVS Result: " + rebill.getAVS() + Environment.NewLine)
                Console.Write("CVV2 Result: " + rebill.getCVV2() + Environment.NewLine)
                Console.Write("Masked Payment Account: " + rebill.getMaskedPaymentAccount() + Environment.NewLine)
                Console.Write("Card Type: " + rebill.getCardType() + Environment.NewLine)
                Console.Write("Authorization Code: " + rebill.getAuthCode() + Environment.NewLine)
            Else
                Console.Write("Transaction Error: " + rebill.getMessage() + Environment.NewLine)
            End If
        End Sub
    End Class
End Namespace